src/pushjack/apns.py

```diff
@@ -764,6 +776,346 @@ class APNSExpiredToken(namedtuple('APNSExpiredToken', ['token', 'timestamp'])):
     pass
 
 
+class HttpStatus(object):
+    OK = 200
+    BAD_REQUEST = 400
+    FORBIDDEN = 403
+    METHOD_NOT_ALLOWED = 405
+    GONE = 410
+    PAYLOAD_TOO_LARGE = 413
+    TOO_MANY_REQUESTS = 429
+    INTERNAL_SERVER_ERROR = 500
+    SERVICE_UNAVAILABLE = 503
+
+
+class AuthTokenBase(object):
+    # Sub-class must assign values to them in __init__.
+    # The only reason why these vars added here is to increase the readability.
+
+    # The date and the time the last encryption was performed.
+    encrypted_at = None
+    team_id = None
+    key_id = None
+    algorithm = None
+    # Base64URL-encoded token in JWT format.
+    _encrypted_token = None
+
+    def value(self):
+        raise NotImplementedError(
+            '{} must be implemented'.format(self.__class__.value.__name__))
+
+    @property
+    def encrypted_token(self):
+        if not self._encrypted_token or not self.is_fresh():
+            self._encrypted_token = self._encrypt()
+
+        return self._encrypted_token
+
+    def _encrypt(self):
+        """
+        Encrypt and return the auth token.
+        """
+        self.encrypted_at = _utc_now()
+
+        # Encrypt it.
+        _token = jwt.encode(
+            payload={
+                'iss': self.team_id,
+                'iat': get_epoch(self.encrypted_at)
```

Comment:
As per above, I think this can just be `datetime.utcnow()`.

Reply:
Yes, you are right. I will correct it.